### ISSUE TITLE
[Tune] Change self.Config in tune.Trainable object

### ISSUE THREAD

**Issue by Jiamian-Wang** [ray-project/ray#21124]
### Search before asking

- [X] I had searched in the [issues](https://github.com/ray-project/ray/issues) and found no similar feature requirement.

### Description

Hi, I'm using the scheuler population based trining (PBT). Currently, for each agent,  the hyperparameters can be modified through exploit/explore process. While they are kept unchanged within in the Trainable class (i.e., learning rate remains the same during training iterations while might be changed only after exploit/explore). How we can optimize the hyperparameters (i.e., in bi-level optimization case) within the Trainable class and update the self.config accordingly? So that the optimized hyperprameter values appear when we recall Trainable class. Currently I update the self.config but it seems not work......

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

**Comment by amogkam** [comment]
Hey @Jiamian-Wang! Do you mind giving an example of what you are trying to do? Do you want to be able to change the hyperparameters for PBT from within the Trainable?
